gateway-provider-security-pac4j/src/main/java/org/apache/knox/gateway/pac4j/session/KnoxSessionStore.java

```diff
@@ -176,6 +211,25 @@ public void set(WebContext context, String key, Object value) {
         }
     }
 
+    /**
+     * Keep only the fileds that are needed for Pac4J.
+     * Used to reduce the cookie size.
+     * @param value profile object
+     * @return trimmed profile object
+     * @since 1.3.0
+     */
+    private Object clearUserProfile(final Object value) {
+        if(value instanceof LinkedHashMap<?,?>) {
```

Comment:
Could this be any Map instead of a LinkedHashMap? Just thinking if profiles Map impl could change later but would still have the same underlying interface.

Reply:
Good point, let me try with a generic Map and see how it works.